packages/store-sync/src/world/getSystemAbi.ts

```diff
@@ -0,0 +1,35 @@
+import { Client, Abi, Address, hexToString, parseAbiItem } from "viem";
+import metadataConfig from "@latticexyz/world-module-metadata/mud.config";
+import { getRecords } from "../getRecords";
+
+export async function getSystemAbi({
+  client,
+  worldAddress,
+  fromBlock,
+  toBlock,
+  indexerUrl,
+  chainId,
+}: {
+  readonly client: Client;
+  readonly worldAddress: Address;
+  readonly fromBlock?: bigint;
+  readonly toBlock?: bigint;
+  readonly indexerUrl?: string;
+  readonly chainId?: number;
+}): Promise<Abi> {
+  const { records } = await getRecords({
+    table: metadataConfig.tables.metadata__ResourceTag,
+    worldAddress,
+    chainId,
+    indexerUrl,
+    client,
+    fromBlock,
+    toBlock,
+  });
+
+  const abi = records
+    .filter(({ tag }) => hexToString(tag).replace(/\0+$/, "") === "abi")
+    .flatMap(({ value }) => (value === "0x" ? [] : hexToString(value).split("\n").map(parseAbiItem)));
```

Comment:
according to docs, `parseAbiItem` accepts an array so this could be 
```suggestion
    .flatMap(({ value }) => (value === "0x" ? [] : parseAbiItem(hexToString(value).split("\n"))));
```

Reply:
Looks like there is a bug as it does not work as described in the docs, will file an issue.

Reply:
Oh just realized what it actually does. It only parses one ABI item but it accepts an array in case the function signature contains a struct. In such case the parser will know how to interpret the struct as well. There might possible be more use cases as well.

Reply:
ahh okay! I misunderstood the docs then 

looks like `parseAbi` does what I was hoping for

Reply:
ah that's right, adjusted